lib/plugins/authldap/auth.php

```diff
@@ -193,16 +197,33 @@ protected function _getUserData($user, $inbind = false) {
             $filter = "(ObjectClass=*)";
         }
 
-        $sr     = $this->_ldapsearch($this->con, $base, $filter, $this->getConf('userscope'));
-        $result = @ldap_get_entries($this->con, $sr);
+		$this->_debug('LDAP Filter: '.htmlspecialchars($filter), 0, __LINE__, __FILE__);
+		
         $this->_debug('LDAP user search: '.htmlspecialchars(ldap_error($this->con)), 0, __LINE__, __FILE__);
         $this->_debug('LDAP search at: '.htmlspecialchars($base.' '.$filter), 0, __LINE__, __FILE__);
-
-        // Don't accept more or less than one response
-        if(!is_array($result) || $result['count'] != 1) {
-            return false; //user not found
+		$sr     = $this->_ldapsearch($this->con, $base, $filter, $this->getConf('userscope'));
+		
+		$result = @ldap_get_entries($this->con, $sr);
+
+        // if result is not an array
+        if(!is_array($result)) {
+			// no objects found
+			$this->_debug('LDAP search returned non-array result: '.htmlspecialchars(print($result)), -1, __LINE__, __FILE__);
+            return false; 
         }
-
+		
+		// Don't accept more or less than one response
+		if ($result['count'] != 1)		{
+			$this->_debug('LDAP search returned '.htmlspecialchars($result['count']).' results while it should return 1!', -1, __LINE__, __FILE__);
+			//for($i = 0; $i < $result["count"]; $i++) {
```

Comment:
I agree. do you want me to remove the code in comment? or you prefer to do it yourself?